homeassistant/components/econet/climate.py

```diff
@@ -35,8 +35,13 @@
     ThermostatOperationMode.OFF: HVACMode.OFF,
     ThermostatOperationMode.AUTO: HVACMode.HEAT_COOL,
     ThermostatOperationMode.FAN_ONLY: HVACMode.FAN_ONLY,
+    ThermostatOperationMode.EMERGENCY_HEAT: HVACMode.HEAT,
+}
+HA_STATE_TO_ECONET = {
+    value: key
+    for key, value in ECONET_STATE_TO_HA.items()
+    if key != ThermostatOperationMode.EMERGENCY_HEAT
 }
```

Comment:
This is problematic.

If you set one state, it would jump to another once an update is received?

Honestly, looking at this, emergency heat should probably be extracted into a separate switch entity.

../Frenck

Reply:
EMERGENCY_HEAT is already a separate switch, which is the correct model for HA, I think. It doesn't represent the thermostat well though, because it's a separate mode. The thermostat is either in HEAT or EMERGENCY_HEAT, not both. But because we can't add climate modes, this is how it must be done. Ideally I'd prefer to see it as a climate model.

And yes, the behavior was that I would set it to HEAT mode, and HA would also set the EMERGENCY_HEAT switch.